libmscore/timesig.cpp

```diff
@@ -314,20 +320,19 @@ void TimeSig::layout()
 
             qreal displ = (numOfLines & 1) ? 0.0 : (0.05 * _spatium);
 
-            //align on the wider
+            // align on the wider
+            // if the other is empty, directly put in the centre of the clef
             qreal pzY = yoff - (denRect.width() < 0.01 ? 0.0 : (displ + numRect.height() * .5));
-            qreal pnY = yoff + displ + denRect.height() * .5;
+            qreal pnY = yoff + (numRect.width() < 0.01 ? 0.0 : (displ + denRect.height() * .5));
```

Comment:
Now I think this's still questionable. Only putting a denominator and no numerator for time signature indeed doesn't make sense, the problem is that we still have to give a reasonable layout if some user does so. I'm not sure whether that layout should be the way it always has been (do not change the position of denominator), or the way with this change (place the denominator in the centre).